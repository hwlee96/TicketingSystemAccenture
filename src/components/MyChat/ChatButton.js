import React, { Component } from "react";
import MyChat from "./MyChat";
import Fab from "@material-ui/core/Fab";

import chatIcon from "./chat.svg";

const style = {
  margin: 0,
  top: "auto",
  right: 20,
  bottom: 20,
  left: "auto",
  position: "fixed",
  backgroundColor: "#416FBD",
  color: "#FFFFFF"
};

class ChatButton extends Component {
  constructor(props) {
    super(props);
    this.state = { show: false };

    this.toggleDiv = this.toggleDiv.bind(this);
  }

  toggleDiv = () => {
    const { show } = this.state;
    this.setState({ show: !show });
  };
  render() {
    return (
      <div>
        <div>
          {this.state.show && (
            <MyChat
              author={this.props.author}
              requestID={this.props.requestID}
            />
          )}
        </div>
        <Fab style={style} onClick={this.toggleDiv}>
          <img src={chatIcon} alt="chatIcon" />
          <i className="fas fa-comment" />
        </Fab>
      </div>
    );
  }
}
export default ChatButton;
