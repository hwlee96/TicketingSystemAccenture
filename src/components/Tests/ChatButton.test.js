import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount, render, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import ChatButton from "../MyChat/ChatButton";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = renderer.create(<ChatButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
