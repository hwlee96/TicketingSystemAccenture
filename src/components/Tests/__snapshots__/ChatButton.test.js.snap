// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div />
  <button
    className="MuiButtonBase-root-11 MuiFab-root-1"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#416FBD",
        "bottom": 20,
        "color": "#FFFFFF",
        "left": "auto",
        "margin": 0,
        "position": "fixed",
        "right": 20,
        "top": "auto",
      }
    }
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiFab-label-2"
    >
      <img
        alt="chatIcon"
        src="chat.svg"
      />
      <i
        className="fas fa-comment"
      />
    </span>
    <span
      className="MuiTouchRipple-root-14"
    />
  </button>
</div>
`;
